app/uk/gov/hmrc/agentservicesaccount/views/pages/desi_details/select_changes.scala.html

```diff
@@ -0,0 +1,85 @@
+@*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *@
+
+@import uk.gov.hmrc.agentservicesaccount.config.AppConfig
+@import uk.gov.hmrc.agentservicesaccount.views.html.main
+@import uk.gov.hmrc.agentservicesaccount.controllers.desiDetails
+@import uk.gov.hmrc.govukfrontend.views.html.components.{GovukErrorSummary, ErrorSummary, GovukButton, GovukCheckboxes, FormWithCSRF}
+@import uk.gov.hmrc.govukfrontend.views.viewmodels.button.Button
+@import uk.gov.hmrc.govukfrontend.views.html.components.{Checkboxes, Fieldset, Legend, Text, CheckboxItem}
+@import uk.gov.hmrc.hmrcfrontend.views.html.components.implicits._
+@import uk.gov.hmrc.govukfrontend.views.Aliases.Hint
+@import uk.gov.hmrc.agentservicesaccount.models.SelectChanges
+
+@this(main: main, h1: h1,
+formWithCSRF: FormWithCSRF,
+govukCheckboxes: GovukCheckboxes,
+govukErrorSummary: GovukErrorSummary,
+govukButton: GovukButton
+)
+
+@(form: Form[SelectChanges])(implicit request: Request[_], msgs: Messages, appConfig: AppConfig)
+
+@main(pageTitle = withErrorPrefix(form.hasErrors, msgs("update-contact-details.select-changes.title")), mimicBrowserBackLink = true, showSecondaryNav = false) {
+
+    @if(form.hasErrors){
+        @govukErrorSummary(ErrorSummary().withFormErrorsAsText(form))
+    }
+
+    @formWithCSRF(action = desiDetails.routes.SelectDetailsController.onSubmit, 'id -> "input") {
+        @govukCheckboxes(Checkboxes(
+            fieldset = Some(Fieldset(
+               legend = Some(Legend(
+                    content = Text(msgs("update-contact-details.select-changes.h1")),
+                    classes = "govuk-fieldset__legend--l",
+                    isPageHeading = true
+                ))
+            )),
+            hint = Some(Hint(
+                content = Text(msgs("update-contact-details.select-changes.hint")))),
+            name = "Select Changes",
+            items = Seq(
+                CheckboxItem(
+                    content = Text(msgs("update-contact-details.select-changes.business-name")),
+                    value = "businessName",
+                    id = Some("businessName"),
+                    name = Some("businessName")
```

Comment:
we are missing a `checked` property which should be able to find values in the form data to determine if this is checked or not, for example when revisiting the page from the CYA

Reply:
so this checked logic can be applied to all the checkboxes (again, this is needed in the absence of .withFormField())
```suggestion
                    name = Some("businessName"),
                    checked = form.data.contains("businessName")
                    
```